packages/gasket-plugin-command/lib/index.d.ts

```diff
@@ -0,0 +1,107 @@
+import type { MaybeAsync, GasketConfig } from '@gasket/core';
+import type { PluginData } from '@gasket/plugin-metadata';
+import type { Command } from 'commander';
+
+export interface GasketCommandDefinition {
+  /* Command id/name */
+  id: string;
+
+  /* Command description */
+  description: string;
+
+  /* Command arguments */
+  args?: Array<GasketArgDefinition>;
+
+  /* Command options */
+  options?: Array<GasketOptionDefinition>;
+
+  /* Command action handler */
+  action: (...args: any[]) => MaybeAsync<void>;
+
+  /* Hide from help output */
+  hidden?: boolean;
+
+  /* Default command to run if no command is provided */
+  default?: boolean;
+}
+
+export interface GasketCommand {
+  command: Command;
+  hidden: boolean;
+  isDefault: boolean;
+}
+
+// Default cannot be used when required is true
+export type GasketArgDefinition = {
+  /* Argument name */
+  name: string;
+
+  /* Argument description */
+  description: string;
+
+  /* Is the argument required */
+  required?: true;
+
+  /* Default value for the argument - never if required is true*/
+  default?: never;
+} | {
+  name: string;
+  description: string;
+  required?: false;
+  default?: any;
+};
+
+export type GasketCommandArg = Array<string | boolean>;
```

Comment:
Am I right in understanding that this is a type just used internally? I'd try to ensure that only types that the consuming developer cares about are exported.